packages/url_launcher/android/src/main/java/io/flutter/plugins/urllauncher/UrlLauncherPlugin.java

```diff
@@ -104,23 +123,37 @@ public void onCreate(Bundle savedInstanceState) {
       webview = new WebView(this);
       setContentView(webview);
       // Get the Intent that started this activity and extract the string
-      Intent intent = getIntent();
-      String url = intent.getStringExtra("url");
-      Boolean enableJavaScript = intent.getBooleanExtra("enableJavaScript", false);
-      Boolean enableDomStorage = intent.getBooleanExtra("enableDomStorage", false);
-      webview.loadUrl(url);
-      if (enableJavaScript) {
-        webview.getSettings().setJavaScriptEnabled(enableJavaScript);
-      }
-      if (enableDomStorage) {
-        webview.getSettings().setDomStorageEnabled(enableDomStorage);
-      }
+      final Intent intent = getIntent();
+      final String url = intent.getStringExtra("url");
+      final boolean enableJavaScript = intent.getBooleanExtra("enableJavaScript", false);
+      final boolean enableDomStorage = intent.getBooleanExtra("enableDomStorage", false);
+      final Bundle headersBundle = intent.getBundleExtra(Browser.EXTRA_HEADERS);
+
+      final Map<String, String> headersMap = extractHeaders(headersBundle);
+      webview.loadUrl(url, headersMap);
+
+      webview.getSettings().setJavaScriptEnabled(enableJavaScript);
+      webview.getSettings().setDomStorageEnabled(enableDomStorage);
+
       // Open new urls inside the webview itself.
       webview.setWebViewClient(
           new WebViewClient() {
+
+            @Override
+            @SuppressWarnings("deprecation")
+            public boolean shouldOverrideUrlLoading(WebView view, String url) {
+              if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
```

Comment:
I'm not sure I understand why are we overriding this to call loadUrl on versions older than Lolipop?

Reply:
You can verify why running the code in different emulators APIs, but the gist of it is that the first method is deprecated and should be used only in older versions of android, at the same time second method only exist in new API versions and they will not call the first one.

Reply:
Thanks for following up!

I'm still not sure I understand - what breaks if we don't override `shouldOverrideUrlLoading` at all?

Reply:
I don't quite remember, but I guess that is so the user can navigate using hyperlinks.
I didn't included the call to `setWebViewClient`, I just fixed the deprecation warning I was getting in Android Studio.